### PR TITLE
Restructure retinanet steps to avoid modifying args

### PR DESCRIPTION
**Pull request by ianstenbit**
Relevant Keras Core [issue](https://github.com/keras-team/keras-core/issues/680)

### CODE REVIEW

File: keras_cv/models/object_detection/retinanet/retinanet.py [hunk 1]
```diff
@@ -453,50 +461,15 @@ def compute_loss(self, x, y, y_pred, sample_weight, **kwargs):
...
-        boxes, classes = self.label_encoder(x, y_for_label_encoder)
-        super_args = args[:-1] + (
-            (
-                x,
-                {"box": boxes, "classification": classes, "unencoded": y},
-            ),
-        )
-
-        return super().train_step(*super_args)
+        return super().train_step(*args, (x, y))
```
Comment (sampathweb) [inline keras_cv/models/object_detection/retinanet/retinanet.py:466]: Aren't we modifying the args here when we call super?  Were you able to test this with the `keras-core` issue?

Comment (ianstenbit) [reply keras_cv/models/object_detection/retinanet/retinanet.py:466]: I did test this using your repro script (here's my [test](https://colab.research.google.com/drive/10FsofZXYVYvIjsTgpBVD66uxI5D-k_7l?resourcekey=0-EVXVxsoKye8TOfXLtSJwIQ&usp=sharing)), and it works just fine (this is the same approach used by YOLOV8 as well)

**Review by sampathweb — APPROVED** [review]

File: keras_cv/models/object_detection/retinanet/retinanet.py (hunk 1 above)
Comment (sampathweb) [reply keras_cv/models/object_detection/retinanet/retinanet.py:466]: LGTM

**Comment by sampathweb** [comment]
LGTM.  Thanks for fixing this quickly.

**Comment by ianstenbit** [comment]
Jax tests [passed](https://console.cloud.google.com/cloud-build/builds;region=global/1fa8faa9-aef9-40cf-8a2e-8bcd2a965f34?project=keras-team-test), I just cancelled an extra one.

**Pull request merged**